src/Symfony/Component/Form/Extension/Core/DataTransformer/DatePointToDateTimeTransformer.php

```diff
@@ -0,0 +1,64 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Form\Extension\Core\DataTransformer;
+
+use Symfony\Component\Clock\DatePoint;
+use Symfony\Component\Form\DataTransformerInterface;
+use Symfony\Component\Form\Exception\TransformationFailedException;
+
+/**
+ * Transforms between a DatePoint object and a DateTime object.
+ *
+ * @implements DataTransformerInterface<DatePoint, \DateTime>
+ */
+final class DatePointToDateTimeTransformer implements DataTransformerInterface
+{
+    /**
+     * Transforms a DatePoint into a DateTime object.
+     *
+     * @param \DatePoint|null $value A DatePoint object
+     *
+     * @throws TransformationFailedException If the given value is not a \DatePoint
+     */
+    public function transform(mixed $value): ?\DateTime
+    {
+        if (null === $value) {
+            return null;
+        }
+
+        if (!$value instanceof DatePoint) {
+            throw new TransformationFailedException('Expected a \DatePoint.');
```

Comment:
This looks like a built-in type, which is not the case.

I suggest:
```suggestion
            throw new TransformationFailedException(\sprintf('Expected a "%s".', DatePoint::class));
```

Reply:
Fixed